addons/website_sale_delivery/static/src/js/website_sale_delivery.js

```diff
@@ -130,21 +359,70 @@ publicWidget.registry.websiteSaleDelivery = publicWidget.Widget.extend({
      * @private
      * @param {Event} ev
      */
-    _onCarrierClick: function (ev) {
-        var $radio = $(ev.currentTarget).find('input[type="radio"]');
-        this._showLoading($radio);
-        $radio.prop("checked", true);
-        var $payButton = $('button[name="o_payment_submit_button"]');
-        $payButton.prop('disabled', true);
-        var disabledReasons = $payButton.data('disabled_reasons') || {};
-        disabledReasons.carrier_selection = true;
-        $payButton.data('disabled_reasons', disabledReasons);
-        dp.add(this._rpc({
+    _onCarrierClick: async function (ev) {
+        var radio = ev.currentTarget.closest('.o_delivery_carrier_select').querySelector('input[type="radio"]');
+        this._showLoading(radio);
+        radio.checked = true;
+        var payButton = document.querySelector('button[name="o_payment_submit_button"]');
+        payButton.disabled = true;
+        var disabledReasons = payButton.getAttribute('disabled_reasons') || {};
+        if(!disabledReasons){
+            payButton.createAttribute('disabled_reasons')
+        }
+        disabledReasons = true;
+        payButton.setAttribute('disabled_reasons', disabledReasons);
+        const data = await dp.add(this._rpc({
+            route: '/shop/update_carrier',
+            params: {
+                carrier_id: radio.value,
+            },
+        }))
+        await Promise.all([this._onClickShowLocations(ev),this._handleCarrierUpdateResult(data),this._disablePayButtonNoPickupPoint(ev)])
+    },
+
+    _disablePayButtonNoPickupPoint : function (ev){
+        var closestDocElement = ev.currentTarget.closest('.o_delivery_carrier_select');
+        var address = closestDocElement.querySelector('.o_order_location').children[2].innerText;
+        var isPickUp = closestDocElement.children[closestDocElement.children.length - 1].children;
```

Comment:
```suggestion
        const isPickUp = closestDocElement.lastChild.children;
```
That said, are we sure the last element will be related to pickup ? It should be better to use `querySelector`